src/xbot/common/math/FieldPose.java

```diff
@@ -0,0 +1,127 @@
+package xbot.common.math;
+
+public class FieldPose {
+
+    private double m;
+    private double b;
+    private ContiguousHeading heading;
+    private final XYPair fieldPosition;
+    
+    public FieldPose(XYPair point, ContiguousHeading heading) {
+        m = degreesToSlope(heading.getValue());
+        this.fieldPosition = point;
+        this.heading = heading;
+        
+        // if we already have the y-intercept, avoid division by 0.
+        if ((point.x == 0) || (Math.abs(m) < 0.01)) {
```

Comment:
Yes, a RobotPose is a combination of a point/heading, as well as the line described by that heading (so that a robot could attempt to get on that line of travel)